### PR TITLE
feat: Add the option to handle  post layer function that need to grow like BatchNorm

### PR DESCRIPTION
**Pull request by TheoRudkiewicz**
For now we only worked with activation function like ReLU that don't change when there input shape change, hence we did not had to make it grow. However function like BatchNorm need to be extended if there input size increase. This PR allow to handle this. 
- in `_apply_output_changes` we use the method `grow` of each module of the post activation function that has a `grow` method
- we introduce a `extended_post_layer_function` that does not depend of the shape to use it in the `extended_forward` , by default it is the same as the `post_layer_function`

### CODE REVIEW

**Comment by TheoRudkiewicz** [comment]
This PR would require some new test before been merged.

File: tests/test_linear_growing_module.py [hunk 1]
```diff
@@ -400,6 +434,64 @@ def setUp(self):
         for bias in (True, False):
             self.demo_layers[bias] = self.create_demo_layers(bias)
 
+    def test_apply_change_with_sized_post_layer_function(self):
```
Comment (sylvchev) [inline tests/test_linear_growing_module.py:437]: The subtests are a very nice addition here to see the different cases.

File: tests/torch_unittest.py [hunk 2]
```diff
@@ -1,15 +1,37 @@
...
+    - assertShapeEqual: check the shape of a torch tensor
+    - assertAllClose: check that two torch tensors are equal up to a tolerance
 """
 
 from unittest import TestCase
 
 import torch
 
 
+class SizedIdentity(torch.nn.Identity):
```
Comment (sylvchev) [inline tests/torch_unittest.py:16]: Nice addition for the tests

**Review by sylvchev — APPROVED** [review]
Let get this merged

File: src/gromo/modules/growing_module.py [hunk 3]
```diff
@@ -530,6 +531,25 @@ def __init__(
 
         self.layer: torch.nn.Module = layer.to(self.device)
         self.post_layer_function: torch.nn.Module = post_layer_function.to(self.device)
+        if extended_post_layer_function is None:
+            self.extended_post_layer_function = post_layer_function
```
Comment (stephane-rivaud) [inline src/gromo/modules/growing_module.py:535]: In case where `extended_post_layer_function` is None, the `post_layer_function` might trigger device related error if not pushed to the right device. Although this should not cause error due to PyTorch handling policy, the suggested modification provides a more robust implementation.
```suggestion
            self.extended_post_layer_function = post_layer_function.to(self.device)
```

Comment (TheoRudkiewicz) [reply src/gromo/modules/growing_module.py:535]: Thank you for pointing this out ! I believe we can also fix it by using `self.post_layer_function` which has already been moved to the correct device (two lines above). I will implement it soon
